src/coreclr/jit/compiler.cpp

```diff
@@ -1849,13 +1849,12 @@ void Compiler::compInit(ArenaAllocator*       pAlloc,
 
     fgInit();
     lvaInit();
+    optInit();
```

Comment:
Seems fine, but presumably none of the optimizer phases run in the inliner-only `Compiler`, right? Was the new checking invoking checks related to optimizer data, during inlining?

Reply:
The loop table checker is enabled after all phases and is supposed to decide when it should run. To do this It was reading `optLoopCount` which was not initialized for inlinee compilers.

Seemed like a good idea to just initialize it always.

Reply:
While we're on the topic, I wondered if I should just make all check enablement explicit and not rely on the the checks themselves to decide when they should run enabled. Or convert the currently guarded checks to run always and internally decide when they actually checked things. So I left them both as is ....

Reply:
Atleast these checks which are always on?

```
 PhaseChecks::CHECK_EH | PhaseChecks::CHECK_LOOPS | PhaseChecks::CHECK_UNIQUE | PhaseChecks::CHECK_PROFILE;
```